s3shell-kafka-sdk/src/main/resources/template/broker.properties

```diff
@@ -0,0 +1,167 @@
+# The maximum size of block cache the broker can use to cache data read from S3, default 100MB
+s3.block.cache.size=104857600
+# The execution interval for stream object compaction, default 60 minutes
+s3.stream.object.compaction.interval.minutes=60
+# The maximum size of stream object allowed to be generated in stream compaction, default 10GB
+s3.stream.object.compaction.max.size.bytes=10737418240
+# The execution interval for stream set object compaction, default 20 minutes
+s3.stream.set.object.compaction.interval.minutes=20
+# The maximum allowed memory consumption for stream set object compaction, default 200MB
+s3.stream.set.object.compaction.cache.size=209715200
+# The minimum time before a stream set object to be force split into multiple stream object, default 120 minutes
+s3.stream.set.object.compaction.force.split.minutes=120
+# The maximum stream set objects allowed to be compacted in one execution, default 500
+s3.stream.set.object.compaction.max.num=500
+# The baseline network bandwidth of the broker in bytes/s, default 100MB/s. This is used to throttle the network usage during compaction
+# and catch up read
+s3.network.baseline.bandwidth=104857600
+# Set to true to enable metrics collection
+s3.telemetry.metrics.enable=false
+# The metrics level to record, supported values are INFO, DEBUG
+s3.telemetry.metrics.level=INFO
+# The metrics exporter type, supported values are otlp, prometheus, log. Use comma to separate multiple exporters.
+s3.telemetry.metrics.exporter.type=otlp
+# The Prometheus HTTP server host and port, if exporter type is set to prometheus
+# s3.metrics.exporter.prom.host=127.0.0.1
+# s3.metrics.exporter.prom.port=9090
+# Set to true to enable exporting tracing data to OTel Collector
+s3.telemetry.tracer.enable=false
+# The OTel Collector endpoint, if exporter type is set to otlp or tracing is enabled
+# s3.telemetry.exporter.otlp.endpoint=http://${your_host_name}:4317
+# Set following configurations for batching
+s3.telemetry.tracer.span.scheduled.delay.ms=1000
+s3.telemetry.tracer.span.max.queue.size=5120
+s3.telemetry.tracer.span.max.batch.size=1024
+# Metrics report interval
+s3.telemetry.exporter.report.interval.ms=5000
+############################# Settings for Auto Balancer #############################
+# The metric reporter to collect and report metrics for Auto Balancer
+metric.reporters=kafka.autobalancer.metricsreporter.AutoBalancerMetricsReporter
+# The network inbound bandwidth in bytes/s, default 100MB/s. Used in NetworkInCapacityGoal and calculation of inbound bandwidth utilization
+autobalancer.reporter.network.in.capacity=104857600
+# The network outbound bandwidth in bytes/s, default 100MB/s. Used in NetworkOutCapacityGoal and calculation of outbound bandwidth utilization
+autobalancer.reporter.network.out.capacity=104857600
+# The reporter interval of Auto Balancer metric reporter in milliseconds, default 10s
+autobalancer.reporter.metrics.reporting.interval.ms=10000
```

Comment:
What are the functions of these three templates?
Do we need to set so many configurations in the template? (Or can we just use the default configuration?)

Reply:
S3PropUtil use these template files. We need this configuration template so that admin tool can start server without specify config path and file. If we not put the config template in classpath, we will not know where to get the template config content.

Reply:
ok, got it.